### PR TITLE
Fix cockroachdb lock table regression

### PR DESCRIPTION
**Pull request by bithavoc**
In the 6.5 release go migrations started using `LOCK TABLE` which is not supported by CockroachDB (2.1.6).

This PR makes sure the the latest changes can still be used with CockroachDB.

**Note:** Let me know if you're interested in having all the tests run on cockroachDB as well.

### CODE REVIEW

File: collection.go [hunk 1]
```diff
@@ -650,7 +650,14 @@ func (c *Collection) begin(db DB) (*pg.Tx, int64, error) {
...
 		_ = tx.Rollback()
-		return nil, 0, err
+		if strings.Contains(err.Error(), "syntax error at or near \"lock\"") {
+			tx, err = db.Begin()
+			if err != nil {
+				return nil, 0, err
+			}
+		} else {
+			return nil, 0, err
+		}
```
Comment (vmihailenco) [inline collection.go]: Please add a comment that `LOCK TABLE` is not supported in CockroackDB. Also you could swap conditions to use early return:

```
if !if strings.Contains() {
    return
}

db.Begin
```

**Comment by vmihailenco** [comment]
>Note: Let me know if you're interested in having all the tests run on cockroachDB as well.

Would be nice if that is not too much trouble.

**Comment by bithavoc** [comment]
@vmihailenco fixed.

The branch is testable in my fork with go modules ` require( github.com/bithavoc/migrations master-fix-lock-table-crdb)` since it needs to ignore stack frames from `/bithavoc/migrations` when interring migration names.

About the CockroachDB support in TravisCI, I'll send it in a separate PR cool?

**Comment by vmihailenco** [comment]
>I'll send it in a separate PR cool?

Sure, thanks

**Pull request merged**